### ISSUE TITLE
Contributing a multistep optimizer for training big NNs/batches

### ISSUE THREAD

**Issue by hoangcuong2011** [tensorflow/tensorflow#35354]
I noticed a high demand in training big NNs/batches is to create a multistep optimizer. This means the optimizer accumulates gradients from batches until it reaches a desire batch and update model parameters. 

This is similar to [this](https://github.com/tensorflow/tensor2tensor/blob/master/tensor2tensor/utils/multistep_optimizer.py), but for file [keras.optimizer_v2](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/optimizer_v2/optimizer_v2.py) instead.

I can contribute to this. However, I noticed by doing this, the code optimizer_v2.py will need to change a lot. I can create a new similar python file (say, multistep_optimizer_v2.py) to avoid changing in optimizer_v2.py, but the optimizer_v2.py code it self is indeed very long.

I thus create a discussion here to see whether you (Tensorflow Team and community) feel it is worthy to do so, as well as how to do that, or any better solution for doing that, if there is.

Best,

**Comment by hoangcuong2011** [comment]
FYI: I actually did investigation carefully, and found that we actually don't need to modify optimizer_v2.py at all. We rather just need to create a new optimizer (similar to and could inherit from Adam, RMSProp, SGD, etc).
This simplifies things a lot, so, again, if you want such a thing, pls let me know.

**Comment by tanzhenyu** [comment]
Thanks for the report! It might be nice to consider this since we're updating the optimizer interface lately. Would you mind emailing this to developers@tensorflow.org with this idea and see what is the response? We might need a public RFC before implementation.
Meanwhile closing it as this is not a real issue.
